AndroidOnChromeOSDragAndDropDemo/app/src/main/java/dev/hadrosaur/draganddropsample/MainActivity.kt

```diff
@@ -0,0 +1,172 @@
+        val mainConstraintLayout = binding.root
+        setContentView(mainConstraintLayout)
+
+        binding.textDragItem.setOnLongClickListener {
+            val textView = it as TextView
+            val dragContent = "Dragged Text: ${textView.text}"
+
+            //Set the drag content and type
+            val item = ClipData.Item(dragContent)
+            val dragData = ClipData(dragContent, arrayOf(MIMETYPE_TEXT_PLAIN), item)
+
+            //Set the visual look of the dragged object
+            //Can be extended and customized. We use the default here.
+            val dragShadow = View.DragShadowBuilder(textView)
+
+            // Starts the drag, note: global flag allows for cross-application drag
+            textView.startDragAndDrop(dragData, dragShadow, null, DRAG_FLAG_GLOBAL)
+
+            false
+        }
+
+        binding.textDropTarget.setOnDragListener(DropTargetListener())
+    }
+
+    inner class DropTargetListener : View.OnDragListener {
+        override fun onDrag(view: View, event: DragEvent): Boolean {
+            return when(event.action) {
+                ACTION_DRAG_STARTED -> {
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)
+                        || event.clipDescription.hasMimeType("image/png")) {
+
+                        // Greenify background colour so user knows this is a target
+                        view.setBackgroundColor(getColor(android.R.color.holo_green_dark))
+                        return true
+                    }
+                    false
+                }
+                ACTION_DRAG_ENTERED -> {
+                    // More intense green background colour when item is over top of target
+                    view.setBackgroundColor(getColor(android.R.color.holo_green_light))
+                    true
+                }
+                ACTION_DRAG_LOCATION -> { true }
+                ACTION_DRAG_EXITED -> {
+                    // Less intense green background colour when item not over target
+                    view.setBackgroundColor(getColor(android.R.color.holo_green_dark))
+                    true
+                }
+                ACTION_DRAG_ENDED -> {
+                    // Back to default colour when drag is not in process
+                    view.setBackgroundColor(getColor(android.R.color.holo_purple))
+                    true
+                }
+                ACTION_DROP -> {
+                    requestDragAndDropPermissions(event) //Allow items from other applications
+                    val item = event.clipData.getItemAt(0)
+
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
+                        if (item.text != null) {
+                            // Plain text item, show it on screen
+                            binding.textDropTarget.setTextSize(COMPLEX_UNIT_SP, 22f)
+                            binding.textDropTarget.text = item.text
```

Comment:
Perhaps preface the text with "Dropped text: " and wrap it in quotes? See also line 55.